src/webhooks/pubsub/slackScores.test.ts

```diff
@@ -0,0 +1,114 @@
+
+describe('slackScores Tests', function () {
+  let getIssueEventsForTeamSpy, postMessageSpy;
+  beforeAll(() => {
+    getIssueEventsForTeamSpy = jest.spyOn(scoresUtils, 'getIssueEventsForTeam');
+    postMessageSpy = jest.spyOn(bolt.client.chat, 'postMessage');
+  });
+  afterEach(() => {
+    jest.clearAllMocks();
+  });
+
+  it('should handle case when no issues are returned', async () => {
+    getIssueEventsForTeamSpy.mockReturnValue([]);
+    await triggerSlackScores(null, null);
+    expect(postMessageSpy).toHaveBeenCalledWith({
+      blocks: [
+        {
+          text: {
+            emoji: true,
+            text: '🗓️ Weekly GitHub Response Times by Team 🗓️',
+            type: 'plain_text',
+          },
+          type: 'header',
+        },
+        {
+          text: {
+            text: `\`\`\`┌────────────────────────────────────────────────┐
+| Team                          │ % on Time      |
+├────────────────────────────────────────────────┤
+| ospo                          | -   (0/0)      |
+| enterprise                    | -   (0/0)      |
+| issues                        | -   (0/0)      |
+| ingest                        | -   (0/0)      |
+| null                          | -   (0/0)      |
+└────────────────────────────────────────────────┘\`\`\``,
+            type: 'mrkdwn',
+          },
+          type: 'section',
+        },
+      ],
+      channel: 'G01F3FQ0T41',
+      text: 'Weekly GitHub Team Scores',
+    });
+  });
+
+  it('should handle case when issues are returned and sorted by best response times', async () => {
+    getIssueEventsForTeamSpy
+      .mockReturnValueOnce([
+        {
+          issue_id: 1,
+          repository: 'routing-repo',
+          product_area: 'One-Team',
+          is_triaged: true,
+        },
+      ])
+      .mockReturnValueOnce([
+        {
+          issue_id: 2,
+          repository: 'routing-repo',
+          product_area: 'Multi-Team',
+          is_triaged: false,
+        },
+      ])
+      .mockReturnValueOnce([
+        {
+          issue_id: 3,
+          repository: 'routing-repo',
+          product_area: 'Test',
+          is_triaged: true,
+        },
+        {
+          issue_id: 4,
+          repository: 'routing-repo',
+          product_area: 'Test',
+          is_triaged: false,
+        },
+      ])
+      .mockReturnValue([]);
+    await triggerSlackScores(null, null);
+    expect(postMessageSpy).toHaveBeenCalledWith({
+      blocks: [
+        {
+          text: {
+            emoji: true,
+            text: '🗓️ Weekly GitHub Response Times by Team 🗓️',
+            type: 'plain_text',
+          },
+          type: 'header',
+        },
+        {
+          text: {
+            text: `\`\`\`┌────────────────────────────────────────────────┐
+| Team                          │ % on Time      |
+├────────────────────────────────────────────────┤
+| ospo                          | 100 (1/1)      |
```

Comment:
What if there's another team with the same percentage, but a different number of issues (ex: team foobar has 2/2, and team bazqux has 2/4)? Proposal: we should order first by % complete descending, then by number of issues descending, then by alphabetical order ascending. So, we would have:

Frank | 100% (2/2)
Grace | 100% (1/1)
Alice | 50% (2/4)
Charlie | 50% (1/2)
Boris | 0% (0/0)
Daniella | 0% (0/0)

Wdyt?

Reply:
I'm with @azaslavsky. In fact, I thought I said this already but maybe not. :o)

Reply:
good idea

Reply:
pushed!